src/js/references/hooks.ts

```diff
@@ -128,3 +130,55 @@ export function useUpdateSourceTypes(
         register,
     };
 }
+
+/**
+ * Check if the logged in account has the passed `right` on the reference detail is loaded for.
+ *
+ * @param reference - The reference to check
+ * @param right - The right to check for (eg. modify_otu)
+ * @returns Whether the right is possessed by the account
+ */
+export function useCheckReferenceRight(reference: Reference, right: string) {
+    const { data: account, isLoading } = useFetchAccount();
+
+    if (isLoading || reference === null) {
+        return;
```

Comment:
```suggestion
        return {hasPermission: false, isLoading: true};
```
Same format for other return statements as well